### PR TITLE
Align with pending -07 changes

### PR DESCRIPTION
**Pull request by chris-wood**
This updates to the pending -07 changes. A diff between the editor's copy and draft -06 [can be found here](https://tools.ietf.org/rfcdiff?url1=https://tools.ietf.org/id/draft-irtf-cfrg-hpke.txt&url2=https://cfrg.github.io/draft-irtf-cfrg-hpke/draft-irtf-cfrg-hpke.txt). The corresponding draft update [is here](https://github.com/cfrg/draft-irtf-cfrg-hpke/pull/195). 

Before merging, I'd like to hear from @rozbb, @dmcardle, @kjacobs-moz, @jedisct1, and @armfazh to confirm that the test vectors match. 

cc @blipp, too!

### CODE REVIEW

**Comment by kjacobs-moz** [comment]
After changing the draft label from "HPKE-06" to "HPKE-07", the vectors* in https://github.com/cfrg/draft-irtf-cfrg-hpke/pull/195 pass in NSS.

\* Tested only DHKEM(X25519, HKDF-SHA256), HKDF-SHA256, [AES-128-GCM, ChaCha20Poly1305]. I've not added support for Export-Only AEAD.

File: format_vectors.py [hunk 1]
```diff
@@ -18,8 +18,12 @@
     "plaintext", "aad", "nonce", "ciphertext",
 ]
 
+ordered_export_keys = [
+    "exporter_context", "L", "exported_value"
+]
+
 encryption_count_keys = [
-    0, 1, 2, 4, 10, 32, 255, 256, 257
+    0, 1, 2, 4,
```
Comment (bifurcation) [inline format_vectors.py:26]: Why are you dropping these?  I would at least test octet roll-over, so something like:

```suggestion
    0, 1, 2, 4, 255, 256
```

File: format_vectors.py [hunk 2]
```diff
@@ -64,7 +68,8 @@ def entry_mode_value(entry):
 aead_idAES128GCM = 0x0001
 aead_idAES256GCM = 0x0002
 aead_idChaCha20Poly1305 = 0x0003
-aeadMap = {aead_idAES128GCM: "AES-128-GCM", aead_idAES256GCM: "AES-256-GCM", aead_idChaCha20Poly1305: "ChaCha20Poly1305"}
+aead_idExportOnly = 0xFFFF
+aeadMap = {aead_idAES128GCM: "AES-128-GCM", aead_idAES256GCM: "AES-256-GCM", aead_idChaCha20Poly1305: "ChaCha20Poly1305", aead_idExportOnly: "Export-Only AEAD"}
```
Comment (bifurcation) [inline format_vectors.py:72]: Nit: Please add some line breaks here.  Pretty sure Python allows it.

File: hpke_test.go [hunk 3]
```diff
@@ -119,17 +118,19 @@ func assertCipherContextEqual(t *testing.T, suite CipherSuite, msg string, lhs,
...
-	assertBytesEqual(t, suite, fmt.Sprintf("%s: %s", msg, "internal AEAD representation"), got, want)
-
 	// Verify that the internal representation of the cipher suite matches.
 	assert(t, suite, fmt.Sprintf("%s: %s", msg, "KEM scheme representation"), lhs.suite.KEM.ID() == rhs.suite.KEM.ID())
 	assert(t, suite, fmt.Sprintf("%s: %s", msg, "KDF scheme representation"), lhs.suite.KDF.ID() == rhs.suite.KDF.ID())
 	assert(t, suite, fmt.Sprintf("%s: %s", msg, "AEAD scheme representation"), lhs.suite.AEAD.ID() == rhs.suite.AEAD.ID())
+
+	// Verify that the internal AEAD object uses the same algorithm and is keyed
+	// with the same key.
+	if lhs.AEADID != AEAD_EXPORT_ONLY {
```
Comment (bifurcation) [inline hpke_test.go:128]: I might reverse the branches here:

```
if lhs.AEADID == AEAD_EXPORT_ONLY { return }
```

File: crypto.go [hunk 4]
```diff
@@ -799,6 +799,28 @@ func (s chachaPolyScheme) NonceSize() int {
...
+
+type exportOnlyScheme struct {
+}
+
+func (s exportOnlyScheme) ID() AEADID {
+	return AEAD_EXPORT_ONLY
+}
+
+func (s exportOnlyScheme) New(key []byte) (cipher.AEAD, error) {
+	panic("Not supported")
```
Comment (bifurcation) [inline crypto.go:813]: You should test these and verify that they throw.

File: format_vectors.py (hunk 1 above)
Comment (chris-wood) [reply format_vectors.py:26]: The test vectors don't generate this many (see latest draft).

File: format_vectors.py (hunk 2 above)
Comment (chris-wood) [reply format_vectors.py:72]: 👍

File: hpke_test.go (hunk 3 above)
Comment (chris-wood) [reply hpke_test.go:128]: 👍

File: crypto.go (hunk 4 above)
Comment (chris-wood) [reply crypto.go:813]: Good call. I'll add a test for that.

**Comment by chris-wood** [comment]
@kjacobs-moz I updated those vectors to include more encryptions (up to a sequence number of 256). Can you please confirm the new vectors pass for you? (Sorry for the trouble!)

**Comment by rozbb** [comment]
Yup, all good

**Pull request merged**